contrib/cli/great_expectations_contrib/package.py

```diff
@@ -0,0 +1,229 @@
+import importlib
+import inspect
+import json
+import logging
+import os
+import sys
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import Any, ClassVar, Dict, List, Optional, Type
+
+from great_expectations.expectations.expectation import Expectation
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# Type alias that will need to be updated to reflect the complex nature of the 'run_diagnostics' return object
+Diagnostics = Dict[str, Any]
+
+
+@dataclass(frozen=True)
+class ExpectationCompletenessCheck:
+    message: str
+    passed: bool
+
+
+@dataclass(frozen=True)
+class ExpectationCompletenessChecklist:
+    experimental: List[ExpectationCompletenessCheck]
+    beta: List[ExpectationCompletenessCheck]
+    production: List[ExpectationCompletenessCheck]
+
+
+@dataclass(frozen=True)
+class PackageCompletenessStatus:
+    concept_only: int
+    experimental: int
+    beta: int
+    production: int
+    total: int
+
+
+@dataclass(frozen=True)
+class RenderedExpectationDetail:
+    name: str
+    tags: List[str]
+    supported: List[str]
+    status: ExpectationCompletenessChecklist
+
+
+@dataclass(frozen=True)
+class Dependency:
+    text: str
+    link: str
+    version: Optional[str]
+
+
+@dataclass(frozen=True)
+class GitHubUser:
+    username: str
+    full_name: Optional[str]
+
+
+class SocialLinkType(Enum):
+    TWITTER = "TWITTER"
+    INSTAGRAM = "INSTAGRAM"
+    LINKEDIN = "LINKEDIN"
+    MEDIUM = "MEDIUM"
+
+
+@dataclass(frozen=True)
+class SocialLink:
+    account_type: SocialLinkType
+    identifier: str
+
+
+@dataclass(frozen=True)
+class DomainExpert:
+    full_name: str
+    social_links: List[SocialLink]
+    picture: str
+
+
+class Maturity(Enum):
+    CONCEPT_ONLY = "CONCEPT_ONLY"
+    EXPERIMENTAL = "EXPERIMENTAL"
+    BETA = "BETA"
+    PRODUCTION = "PRODUCTION"
+
+
+@dataclass(frozen=True)
+class GreatExpectationsContribPackage:
+    # Core
+    package_name: Optional[str] = None
+    icon: Optional[str] = None
+    description: Optional[str] = None
+    expectations: Optional[List[RenderedExpectationDetail]] = None
+    expectation_count: Optional[int] = None
+    dependencies: Optional[List[Dependency]] = None
+    maturity: Optional[Maturity] = None
+    status: Optional[PackageCompletenessStatus] = None
+
+    # Users
+    owners: Optional[List[GitHubUser]] = None
+    contributors: Optional[List[GitHubUser]] = None
+    domain_experts: Optional[List[DomainExpert]] = None
+
+    # Metadata
+    version: Optional[str] = None
+
+    # Existing configuration
+    config_file: ClassVar[str] = ".great_expectations_package.json"
+
+    def _to_json_file(self) -> None:
```

Comment:
Non-blocking comment: I'd appreciate guidance from @donaldheppner on whether this is a pattern we want to continue to use. If so, should we build it into a base class like `DictDot`? Or would it be better to standardize on something like marshmallow?

Reply:
My preference would be for all config objects to have a "Config" suffix, and that the serialization mechanism live closer to the point of serialization (ie: Marshmallow schema objects should generate Config objects post_dump, and stores own reading and writing to store backends)